chugbot/attendance/index.php

```diff
@@ -0,0 +1,238 @@
+            " Successfully Updated</h5>People taking and viewing attendance will now see the " . chug_term_singular . " for the newly set " . block_term_plural . "</div></div>";
+            echo $successMessage;
+    
+        }
+        
+    }
+    else if (test_post_input("form") == "purge") {
+        // delete old records
+
+        // 1: verify date is valid form
+        $dateInput = test_post_input("date");
+        $fullDate = strtotime($dateInput);
+        $date = date('Y-m-d', $fullDate);
+        if($fullDate == "") {
+            echo "<div class=\"col-md-6 offset-md-3\"><div class=\"alert alert-danger alert-dismissible fade show m-2\" role=\"alert\">" . 
+            "<button type=\"button\" class=\"btn-close\" data-bs-dismiss=\"alert\" aria-label=\"Close\"></button><h5><strong>Error:</strong> " . 
+            "Invalid Date</h5>An invalid date was provided. Please try again to purge old attendance records</div></div>";
+            exit();
+        }
+        else {
+            // 2: do deletions
+            $localErr = "";
+            $dbc = new DbConn();
+            // 2a: delete attendance records
+            $sql = "DELETE FROM attendance WHERE date < '$date'";
+            $result = $dbc->doQuery($sql, $localErr);
+            if ($result == false) {
+                echo dbErrorString($sql, $localErr);
+                exit();
+            }
+            // 2b: delete that attendance was taken for that perek
+            $sql = "DELETE FROM chug_attendance_taken WHERE date < '$date'";
+            $result = $dbc->doQuery($sql, $localErr);
+            if ($result == false) {
+                echo dbErrorString($sql, $localErr);
+                exit();
+            }
+            // show success message
+            $successMessage = "<div class=\"col-md-6 offset-md-3\"><div class=\"alert alert-success alert-dismissible fade show m-2\" role=\"alert\">" . 
+            "<button type=\"button\" class=\"btn-close\" data-bs-dismiss=\"alert\" aria-label=\"Close\"></button><h5>Old Attendance Records Successfully Purged</h5>" .
+            "The earliest day with saved attendance records is <strong>" . date("D F j, Y", strtotime($date)) . "</strong></div></div>";
+            echo $successMessage;
+        }
+    }
+}
+
+// get current chug group - active block matches
+$db = new dbConn();
+$db->addSelectColumn("group_id");
+$db->addSelectColumn("active_block_id");
+$result = $db->simpleSelectFromTable("chug_groups", $err);
+if ($result == false) {
+    echo dbErrorString($sql, $localErr);
+    exit();
+}
+while ($row = mysqli_fetch_array($result, MYSQLI_NUM)) {
+    // row[0] is chug group id, row[1] is active block id
+    // genPickList (used below) pre-selects options based on if the key (block id) maps to a value; pass in array corresponding to that group
+    $group2ActiveBlock[$row[0]] = array($row[1]=>$row[1]);
+}
+
+?>
+
+<div class="card card-body mt-3 mb-3 container">
```

Comment:
Yes, mutually exclusive - one is specifically for assigning which block is active for each chug group, and the other is for the purge function. I think it's easier to keep them together as they're both relatively straightforward, short things which only the admin should be able to do as the "admin settings" for the attendance system. I just feel it would be kind of empty and unnecessarily require extra clicks if we separate them onto different pages.

You need to be logged in as the admin to have access to this page :)